### ISSUE TITLE
server: column "index_recommendations" does not exist in 22.2/22.1 mixed version

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#91346]
In a mixed-version 22.2/22.1 cluster, 22.2 nodes throw lots of errors because they expect a column `index_recommendations` which does not exist yet -- this will only be added by the `AlterSystemStatementStatisticsAddIndexRecommendations` migration when the 22.2 upgrade is finalized. I believe this breaks certain DB Console pages. We need to respect the version gate here.

```
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375  ‹combined-stmts-by-interval›: column ‹"index_recommendations"› does not exist
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +(1) attached stack trace
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  -- stack trace:
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/sql/internal.go:892
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/sql/internal.go:471
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/sql/internal.go:522
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server.collectCombinedStatements
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:180
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server.getCombinedStatementStats
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:76
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server.(*statusServer).CombinedStatementStats
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/combined_statement_stats.go:53
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server.(*statusServer).Statements
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/statements.go:32
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_Statements_Handler.func1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/k8-opt/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:8785
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:114
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1117
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:263
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1120
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:232
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:341
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:230
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.chainUnaryInterceptors.func1.1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1120
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.chainUnaryInterceptors.func1
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1122
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_Statements_Handler
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    github.com/cockroachdb/cockroach/pkg/server/serverpb/bazel-out/k8-opt/bin/pkg/server/serverpb/serverpb_go_proto_/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:8787
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.(*Server).processUnaryRPC
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1283
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.(*Server).handleStream
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:1620
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  | google.golang.org/grpc.(*Server).serveStreams.func1.2
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +  |    google.golang.org/grpc/external/org_golang_google_grpc/server.go:922
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +Wraps: (2) ‹combined-stmts-by-interval›
E221105 11:38:25.194496 310193 server/combined_statement_stats.go:242 ⋮ [n1,status] 375 +Wraps: (3) candidate pg code: 42703
```

Jira issue: CRDB-21216

**Comment by erikgrinaker** [comment]
The fix didn't make it into 22.2.0, unfortunately, it will be in 22.2.1. The problem will go away once the upgrade is finalized (happens automatically by default). If you haven't started finalization yet, you can also consider downgrading to 22.1 in the meanwhile.

**Comment by erikgrinaker** [comment]
If you haven't disabled auto-finalization, then finalization should begin once all nodes are running 22.2. You can see if any migration jobs are still running via e.g. `SHOW JOBS` or in the DB console's jobs page. You can also check your logs to see if there were any problems with any upgrade migrations.
